format/spec.md

```diff
@@ -178,6 +178,11 @@ A **`list`** is a collection of values with some element type. The element field
 
 A **`map`** is a collection of key-value pairs with a key type and a value type. Both the key field and value field each have an integer id that is unique in the table schema. Map keys are required and map values can be either optional or required. Both map keys and map values may be any type, including nested types.
 
+### Semi-structured Types
+
+A **`variant`** is a type to represent semi-structured data. A variant value can store a value of any other type, including `null`, any primitive, struct, list or map value. The variant encoding is defined the [Apache Parquet Project](https://github.com/apache/parquet-format/blob/4f208158dba80ff4bff4afaa4441d7270103dff6/VariantEncoding.md). Variant type is added in [v3](#version-3).
+Limitation: map value with only string-type keys is supported in variant.
```

Comment:
I'm a bit unclear about this. Are we saying that the variant type only supports map values where all the keys must be string-type, and it doesn't support maps with non-string keys?

Reply:
It's not only for Iceberg. In the spec (https://github.com/apache/parquet-format/blob/4f208158dba80ff4bff4afaa4441d7270103dff6/VariantEncoding.md), we assume that the keys in an object are string types: `A field_id is an index into the dictionary in the metadata.` .

So we don't support maps with non-string keys.

Reply:
@flyrain  I updated to the following. Can you take a look?

map values must have string-type keys to be stored in a variant; otherwise, they are not supported.